### PR TITLE
feat: support series items method

### PR DESCRIPTION
**Pull request by jialuoo**
See b/372343841.

### CODE REVIEW

File: bigframes/series.py [hunk 1]
```diff
@@ -1117,6 +1117,15 @@ def ne(self, other: object) -> Series:
         # TODO: enforce stricter alignment
         return self._apply_binary_op(other, ops.ne_op)
 
+    def items(self):
+        column_ids = self._block.value_columns
+        assert (
+            len(column_ids) == 1
+        ), f"Expected lenght of column ids to be 1, but got {len(column_ids)}."
+        bpd_series = bigframes.series.Series(self._block.select_column(column_ids[0]))
+        for index, value in zip(bpd_series.index, bpd_series.values):
```
Comment (TrevorBergeron) [inline bigframes/series.py:1126]: I would expect this to generate two separate queries, each for index and values. This is not quite ideal. Maybe take a look at dataframe.iterrows and copy that pattern.

**Review by TrevorBergeron — CHANGES_REQUESTED** [review]

File: bigframes/series.py (hunk 1 above)
Comment (jialuoo) [reply bigframes/series.py]: Per offline discussion, the query will be based on the to_pandas_batches method in the block.

File: tests/system/small/test_series.py [hunk 2]
```diff
@@ -213,6 +213,30 @@ def test_series_construct_from_list_escaped_strings():
...
+)
+def test_series_items(data, index):
+    bf_series = series.Series(data, index=index)
+    pd_series = pd.Series(data, index=index)
+
+    for (bf_index, bf_value), (pd_index, pd_value) in zip(
+        bf_series.items(), pd_series.items()
+    ):
+        # TODO(jialuo): Remove the if conditions after b/373699458 is addressed.
+        if not pd.isna(bf_index) or not pd.isna(pd_index):
```
Comment (jialuoo) [inline tests/system/small/test_series.py:234]: We still need if clause here to handle edge cases that include types like None and numpy.nan.

**Review by TrevorBergeron — APPROVED** [review]
